integrationTests/chainSimulator/staking/delegation_test.go

```diff
@@ -750,6 +764,358 @@ func testChainSimulatorCreateNewDelegationContract(t *testing.T, cs chainSimulat
 	require.Equal(t, blsKeys[0], hex.EncodeToString(unStakedKeys[0]))
 }
 
+func TestChainSimulator_MaxDelegationCap(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    30,
+	}
+
+	// Test scenario done in staking 3.5 phase (staking v4 is not active)
+	// 1. Add a new validator private key in the multi key handler
+	// 2. Set the initial state for the owner and the 3 delegators
+	// 3. Create a new delegation contract with 1250 egld
+	// 4. Add node to the delegation contract
+	// 5. Delegate from user A 1250 EGLD each, check the topup is 2500
+	// 6. Delegate from user B 501 EGLD each, check it fails
```

Comment:
Please state on point 3. that we set a maximum cap of xxx EGLD, otherwise this is hard to understand :)

Reply:
added